RevenueCatUI/Purchasing/PurchaseHandler.swift

```diff
@@ -114,17 +114,29 @@ extension PurchaseHandler {
         self.track(.impression(eventData))
     }
 
-    func trackPaywallClose(_ eventData: PaywallEvent.Data) {
-        self.track(.close(eventData))
+    /// - Returns: whether the event was tracked
+    @discardableResult
+    func trackPaywallClose() -> Bool {
+        guard let data = self.eventData else {
+            Logger.warning(Strings.attempted_to_track_event_with_missing_data)
+            return false
+        }
+
+        self.track(.close(data.withCurrentDate()))
+        self.eventData = nil
```

Comment:
I'd add another test to verify that cancelling purchases doesn't reset this (so you can track `close` after a purchase cancels).